external-crates/move/crates/move-vm-runtime/src/natives/extensions.rs

```diff
@@ -3,7 +3,56 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use better_any::{Tid, TidAble, TidExt};
-use std::{any::TypeId, collections::HashMap};
+use std::{
+    any::TypeId,
+    cell::{Ref, RefCell, RefMut},
+    collections::HashMap,
+    ops::{Deref, DerefMut},
+    rc::Rc,
+};
+
+/// A helper wrapper around a `Tid`able type that encapsulates interior mutability in a single-threaded
+/// manner.
+///
+/// Note that this is _not_ threadsafe. If you need threadsafe access to the `T` you will need to
+/// handle that within `T'`s type (just like in the previous implementation of the
+/// `NativeContextExtensions`).
```

Comment:
I am a little confused. It appears we are making it so that all mutability needs to be explicit when building up the native context extension, whereas before everything was arbitrarily mutable. What is the motivation for this change?

Reply:
In the discussions we had about how we wanted to change this, the consensus that I gathered is that we wanted to move to this model where the native context extensions do not make any guarantees about mutability, or directly support interior mutability itself, but instead the decision around how to (or not to) support any type of mutability would be a per-extension decisions, and the extension would need to be explicit about how, and if, it will support mutability. 

I think this is generally a better place to be as it makes it much clearer from the perspective of a user of the native extensions about how/if you want to support mutability for the extension (do you want a `Mutex` for some reason? Maybe no mutability, etc).